lib/Doctrine/DBAL/Driver/PDOSqlite/Driver.php

```diff
@@ -66,6 +66,9 @@ protected function _constructPdoDsn(array $params)
             $dsn .= $params['path'];
         } elseif (isset($params['memory'])) {
             $dsn .= ':memory:';
+            if (isset($params['cache'])) {
```

Comment:
Possibly a better idea to pass in a pre-built PDO instance, when using advanced driver settings?

Reply:
Or maybe accept a pre-built DSN? I don't like the idea of passing a PDO instance since it breaks encapsulation (therefore, unexpected side effects are possible) and is not supported by non-PDO drivers.

Reply:
Yeah, I think it's better to not have application code touching PDO directly - passing the dsn seems cleaner to me.

```php
            'connection' => [
//                'url' => 'sqlite:///:memory:?cache=shared',
                'driver' => 'pdo_sqlite',
                'dsn' => 'sqlite::memory:?cache=shared',
            ],
```